modules/system/console/OctoberUtil.php

```diff
@@ -339,7 +340,26 @@ protected function utilPurgeOrphans()
             return;
         }
 
-        // @todo
+        $isLocalStorage = Config::get('cms.storage.uploads.disk', 'local') === 'local';
+
+        $orphanedFiles = FileModel::whereNull('attachment_id')->orWhereNull('attachment_type')->delete();
+        $orphanedFiles += FileModel::whereDoesntHaveMorph('attachment', '*')->delete();
```

Comment:
Doesn't this query do the same thing as the previous one as well as evaluating if there's actually related records? Also, calling `->delete()` on the query builder will not delete the underlying files themselves, the delete() method on the model needs to be called instead.

Reply:
No, the second query doesn't catch all the cases.

I was looping through at some point and you questioned this method @LukeTowers ... !?

Reply:
You were looping through every single file on the system and making a separate DB call for each file to attempt to load the relationship. If you had only 1000 records in the system this would load 1000-2000 separate models and make 1001 separate DB queries.

Can you do `FileModel::whereDoesntHaveMorph('attachment', '*')->orWhereNull('attachment_id')->orWhereNull('attachment_type')->get()` and then loop over those results calling delete()?

Reply:
The problem with this is that I will still need to go through each file model if the "--missing-files" option is provided...